cmd/update-main_test.go

```diff
@@ -158,6 +164,59 @@ func TestGetCurrentReleaseTime(t *testing.T) {
 	}
 }
 
+// Tests user agent string.
+func TestUserAgent(t *testing.T) {
+	customUserAgent := func(mode string, extraMode string) string {
```

Comment:
I would recommend to have a constant string comparison over string generation functions.  This avoids making test code breakage.